akka-actor-typed/src/main/scala/akka/actor/typed/internal/receptionist/LocalReceptionist.scala

```diff
@@ -102,13 +110,24 @@ private[akka] object LocalReceptionist extends ReceptionistBehaviorProvider {
     def onCommand(ctx: ActorContext[Any], cmd: Command): Behavior[Any] = cmd match {
       case ReceptionistMessages.Register(key, serviceInstance, maybeReplyTo) =>
         ctx.log.debug2("Actor was registered: {} {}", key, serviceInstance)
-        watchWith(ctx, serviceInstance, RegisteredActorTerminated(key, serviceInstance))
+        val watcher = watchWith(ctx, serviceInstance, RegisteredActorTerminated(key, serviceInstance))
```

Comment:
`watchWith` has the comment `FIXME #26505: replace by simple map in our state`. Maybe worth fixing that now since we are anyway adding a Map?

Reply:
If I understood it correctly that map would contain `ActorRef[_] -> Set[ServiceKey[_]]` to keep track of what services stopped if an actor dies. I think that will simplify things.